src/EditorFeatures/Core/Tagging/AsynchronousViewTaggerProvider.cs

```diff
@@ -13,25 +13,20 @@
 
 namespace Microsoft.CodeAnalysis.Editor.Tagging
 {
-    internal abstract class AsynchronousViewTaggerProvider<TTag> : AbstractAsynchronousTaggerProvider<TTag>,
-        IViewTaggerProvider
+    internal abstract class AsynchronousViewTaggerProvider<TTag> : AbstractAsynchronousTaggerProvider<TTag>, IViewTaggerProvider
         where TTag : ITag
     {
-        protected AsynchronousViewTaggerProvider(
-            IThreadingContext threadingContext,
-            IAsynchronousOperationListener asyncListener,
-            IForegroundNotificationService notificationService)
-                : base(threadingContext, asyncListener, notificationService)
+        protected AsynchronousViewTaggerProvider(IThreadingContext threadingContext, IAsynchronousOperationListener asyncListener)
+            : base(threadingContext, asyncListener)
         {
         }
 
-        // TypeScript still is moving to calling the new constructor that takes an IThreadingContext. Until then, we can fetch one from another service of ours that
-        // already does. When TypeScript moves calling the new constructor, this should be deleted.
+        // TypeScript still is moving to calling the new constructor that does not take a IForegroundNotificationService.
+        // Until then, we can fetch one from another service of ours that already does. When TypeScript moves calling the
+        // new constructor, this should be deleted.
         [Obsolete("This overload exists for TypeScript compatibility only and should not be used in new code.")]
-        protected AsynchronousViewTaggerProvider(
-            IAsynchronousOperationListener asyncListener,
-            IForegroundNotificationService notificationService)
-                : this(((Implementation.ForegroundNotification.ForegroundNotificationService)notificationService).ThreadingContext, asyncListener, notificationService)
+        protected AsynchronousViewTaggerProvider(IThreadingContext threadingContext, IAsynchronousOperationListener asyncListener, IForegroundNotificationService _)
+            : base(threadingContext, asyncListener)
         {
         }
```

Comment:
based on current state of TS code here: https://devdiv.visualstudio.com/DevDiv/_git/TypeScript-VS?path=%2FVS%2FLanguageService%2FTypeScriptLanguageService%2FFeatures%2FClassifier%2FAbstractAsyncClassificationTaggerProvider.cs&_a=contents&version=GBmain